### ISSUE TITLE
Web - Search-Redirected back to "Expenses" when refreshing the page while in "chats"

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48817]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.31-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N/A
**Email or phone of affected tester (no customers):** applausetester+shsb22ertt@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Log into newdot with any account
2. Go to Search option from bottom tab
3. Click on "Chat" type from LHP
4. Click on any message from the rendered results
5. When the report view is opened in RHP refresh the browser

## Expected Result:
App stays in "Chat" search type

## Actual Result:
App navigates to "Expense" search type in the background while the RHP of the report view is still opened

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/79c26c9d-4457-4582-8ebc-623d82633771

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-control

**Comment by luacmartins** [comment]
I can take this since it's part of Search

**Comment by luacmartins** [comment]
cc @shubham1206agra

**Comment by luacmartins** [comment]
This also happens with the current data types, e.g. Invoice, Trips so demoting to NAB. @Kicu @WojtekBoman

**Comment by Kicu** [comment]
Yup it is quite clear why this behaviour happens, based on how search + navigation works. 

I think we'd need to think a bit how best to improve this behaviour if we want to.

In general we are basing the whole state of search on current `q` parameter in URL. This has been great so far - single source of truth and we have 0 bugs related to synchronisation between query and other parts of search.

But this also means behaviours like this one, where if we don't have query in URL then we loose the previous state. I'm a bit reluctant to change this behaviour for now. 🤔

**Comment by luacmartins** [comment]
I agree, but I think that's something we'll eventually have to do to allow for a better navigation experience. We don't need to rush though and can think more carefully on how to solve this without creating bugs.

**Comment by johncschuster** [comment]
Thanks for the thoughtful discussion @Kicu and @luacmartins! What's the next step here? Do we need to take this to Slack to discuss in more detail?

**Comment by luacmartins** [comment]
@Kicu let me know if you or someone at SWM can take on this issue.

**Comment by Kicu** [comment]
@luacmartins but what is the expectation. I described that this is kinda complex and we should think about how to best solve it. Then you agreed.
And now what do we want to do? should I just come up with a solution that will allow us to always go back?

**Comment by luacmartins** [comment]
Oh I just meant to take this on and investigate a good holistic solution for this.

**Comment by Kicu** [comment]
Alright makes sense.
Please assign me to the issue then :)

**Comment by ikevin127** [comment]
Not overdue, this is being worked on.

**Comment by Kicu** [comment]
I will have some update on this issue today or tomorrow, I was handling some other search tasks last week

**Comment by Kicu** [comment]
@luacmartins 
So I consulted with both @adamgrzybowski and @WojtekBoman and our general consensus is that we will use `backTo` param to handle this case.

Correctly set and handled `backTo` will take care of both: having proper screen underneath the modal and going back to correct screen on back button click.

I should have the PR ready tomorrow, it will be relatively small so I can do it in between other 2.4 tasks

**Comment by Kicu** [comment]
calm down melvin ☮️ this will have a PR opened today

**Comment by Kicu** [comment]
PR is open for review

**Comment by ikevin127** [comment]
⚠️ Automation failed here -> this should be on **[HOLD for Payment 2024-10-07]** according to today’s production deploy from https://github.com/Expensify/App/pull/49641#issuecomment-2384303583.

cc @johncschuster

**Comment by johncschuster** [comment]
Thanks for the ping, @ikevin127! I've updated the title and will issue payment after the regression window is passed! 👍

**Comment by johncschuster** [comment]
# Payment Summary:

Contributor+: @ikevin127 paid $250 via [Upwork](https://www.upwork.com/nx/wm/offer/104302485)
Contributor: @Kicu - No payment due

Upwork job [here](https://www.upwork.com/nx/wm/offer/104302485)! Please apply!

@ikevin127 / @Kicu, can you provide the regression test steps if you feel we need them? Thanks!

**Comment by ikevin127** [comment]
@johncschuster Offer accepted. Sure, I will provide the BZ Checklist today 👍

**Comment by johncschuster** [comment]
Thanks for confirming that, @ikevin127! Please post the test steps as soon as possible. Regarding payment, I've issued it now, and wanted to leave a note here since I've heard in Slack we're having trouble with payments on Upwork's side. ([Slack](https://expensify.slack.com/archives/C01GTK53T8Q/p1728319262830679))

Can you please confirm whether or not you've received the payment? I will wait to hear from you before I end the contract.

<img width="775" alt="2024-10-07_16-25-42" src="https://github.com/user-attachments/assets/ffe81d3f-0d4f-4b89-85ca-ac56fe23c025">

**Comment by ikevin127** [comment]
@johncschuster Yes, I confirm that I received the payment ($250). Thank you!

I will post the BZ Checklist in the next hour.

**Comment by ikevin127** [comment]
### Regression Test Proposal

1. Go to Search option from bottom tab.
2. Click on "Chat" type from LHP or use custom advanced filters.
3. Click on any item from the rendered results.
4. When the report view is opened in RHP, refresh the browser.
5. Verify that the report open behind the RHP is the same after browser refresh as before.
6. Verify that the RHP back button works.

Do we agree 👍 or 👎.

cc @johncschuster
